### PR TITLE
Move AppStatusHandler to StatusCoreComponents

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- now with the newly added StatusCoreComponents, AppStatusHandler belongs there
  
## Proposed changes
- moved the components, no logic changed
- completely removed statusHandlerUtilsFactory

## Testing procedure
- standard system test with upgrade
- query a node before upgrade on `/node/status`, then query it again after upgrade. It should have the same metrics after upgrade
- check following metrics are persistent after restarts:
`erd_count_consensus, erd_count_consensus_accepted_blocks, erd_count_accepted_blocks, erd_count_leader, erd_num_transactions_processed, erd_num_shard_headers_processed, erd_nonce, erd_current_round, erd_nonce_at_epoch_start, erd_round_at_epoch_start, erd_total_supply, erd_total_fees, erd_dev_rewards, erd_inflation, erd_epoch_for_economics_data, erd_accounts_snapshot_num_nodes`

## Pre-requisites

Based on the [Contributing Guidelines](CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: factory/interface.go [hunk 1]
```diff
@@ -520,3 +522,9 @@ type TrieSyncStatisticsProvider interface {
 	ProcessingTime() time.Duration
 	NumIterations() int
 }
+
+// PersistenStatusHandler defines a persistend status handler
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply factory/interface.go:526]: fixed

File: factory/statusCore/statusCoreComponents.go [hunk 2]
```diff
@@ -46,10 +98,53 @@ func (sccf *statusCoreComponentsFactory) Create() (*statusCoreComponents, error)
...
+
+	persistentHandler, err := persister.NewPersistentStatusHandler(sccf.coreComp.InternalMarshalizer(), sccf.coreComp.Uint64ByteSliceConverter())
+	if err != nil {
+		return nil, err
+	}
+	appStatusHandlers = append(appStatusHandlers, persistentHandler)
+	if len(appStatusHandlers) > 0 {
+		handler, err = statusHandler.NewAppStatusFacadeWithHandlers(appStatusHandlers...)
+		if err != nil {
+			log.Warn("cannot init AppStatusFacade", "error", err)
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply factory/statusCore/statusCoreComponents.go:114]: correct, updated

File: factory/statusCore/statusCoreComponents.go [hunk 3]
```diff
@@ -46,10 +98,53 @@ func (sccf *statusCoreComponentsFactory) Create() (*statusCoreComponents, error)
...
+		handler = statusHandler.NewNilStatusHandler()
+		log.Debug("no AppStatusHandler used: started with NilStatusHandler")
+	}
+
+	err = metrics.InitBaseMetrics(handler)
+	if err != nil {
+		return nil, err
+	}
+
+	err = metrics.InitConfigMetrics(handler, sccf.epochConfig, sccf.economicsConfig, sccf.coreComp.GenesisNodesSetup())
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply factory/statusCore/statusCoreComponents.go:126]: done

**Review by bogdan-rosianu — DISMISSED** [review]

File: integrationTests/testConsensusNode.go [hunk 4]
```diff
@@ -35,6 +35,7 @@ import (
 	"github.com/ElrondNetwork/elrond-go/testscommon"
 	"github.com/ElrondNetwork/elrond-go/testscommon/cryptoMocks"
 	dataRetrieverMock "github.com/ElrondNetwork/elrond-go/testscommon/dataRetriever"
+	factory2 "github.com/ElrondNetwork/elrond-go/testscommon/factory"
```
Comment (iulianpascalau) [inline integrationTests/testConsensusNode.go:38]: rename to testFactory?

**Review by iulianpascalau — COMMENTED** [review]
Good cleanup :+1:

**Review by bogdan-rosianu — APPROVED** [review]

**Review by iulianpascalau — DISMISSED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Review by bogdan-rosianu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**